lua/orgmode/treesitter/headline.lua

```diff
@@ -475,6 +475,28 @@ function Headline:parse(pattern)
   return matching_nodes[1], match
 end
 
+---Return the line number where content can be appended within
+---the drawer with the given name
+---@param name string
+---
+---@return number
+function Headline:get_drawer_append_line(name)
+  local section = self.headline:parent()
+  local q = tree_utils.parse_query('(body (drawer) @drawer)')
```

Comment:
Let's extract this part to a separate method called `Headline:get_drawer(name)`, which would find the drawer node by the given name, case-insensitively.  Then here you can find the drawer with that method and return the appropriate line number. You can closely mimic how `properties()` and `set_property` methods work.

Reply:
Something like this?